### DISCUSSION TITLE
Including pyg in setup.py file

### DISCUSSION THREAD

**Discussion by SamWright822** [pyg-team/pytorch_geometric#5082] [Q&A] [answered]
Hi! I am trying to include pytorch geometric installs in a setup.py file, which includes the following:

        install_requires=[
            "torch==1.12.0",
            "torch-scatter@https://data.pyg.org/whl/torch-1.12.0+cpu.html",
            "torch-sparse@https://data.pyg.org/whl/torch-1.12.0+cpu.html",
            "torch-cluster@https://data.pyg.org/whl/torch-1.12.0+cpu.html",
            "torch-spline-conv@https://data.pyg.org/whl/torch-1.12.0+cpu.html",
            "torch-geometric@https://data.pyg.org/whl/torch-1.12.0+cpu.html",
            "torch-spline-conv@https://data.pyg.org/whl/torch-1.12.0+cpu.html",
        ],

I am getting this error, however:

    Collecting torch==1.12.0
      Downloading torch-1.12.0-cp38-cp38-manylinux1_x86_64.whl (776.3 MB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 776.3/776.3 MB 157.1 MB/s eta 0:00:00
    Collecting torch-scatter@ https://data.pyg.org/whl/torch-1.12.0+cpu.html
      Downloading https://data.pyg.org/whl/torch-1.12.0+cpu.html (7.3 kB)
      ERROR: Cannot unpack file /tmp/pip-unpack-pkc6wpw3/torch-1.12.0+cpu.html (downloaded from /tmp/pip-install-d7wq8x2q/torch-scatter, content-type: text/html); cannot detect archive format
    ERROR: Cannot determine archive format of /tmp/pip-install-d7wq8x2q/torch-scatter

Is there another way to include pyg in the setup files?

**Comment by rusty1s** [comment, marked as answer]
I don't think this is possible since `setup.py` will only trust official sources such as pypi. The only way to overcome this is via `requirements.txt` files or via `conda`, e.g.
```
-f https://data.pyg.org/whl/torch-1.12.0+cpu.html
torch==1.12.0
torch-scatter
torch-sparse
...
```

**Reply by SamWright822** [reply]
Perfect, thank you so much! :)
